wdl/qc_imputation.wdl

```diff
@@ -23,44 +22,64 @@ workflow qc_imputation {
 
     File? exclude_denials
     File? duplicate_samples
-    ## could compute a single PCA for all samples in panel and across batches and pass the PCs forward to per chr qc.
-    ## Now the PCA uses only single chromosome variants in chr_qc... probably not gonna make a huge difference but that should be the correct way.
-    ## it is not crucial/detrimental though as failing to account for structure in panel comparison will only remove variants that might be region specific
-    ## but won't introduce bad variants.
-
-    ## We don't want to use bad variants or high ld regions as covariates when comparing variant freq to panel, so added qc
-    ## at this level. Also PCA fails if very high missingness individuals in...
 
     ### symmetric variants close to 0.5 can remain with other data than FinnGen
     # (legacy data aligned with Will Raynards mapping files according to Timo and Affy data should be aligned correctly??)
     #
     ## ... Maybe better remove symmetric MAF > 0.4 etc. at some point? Mitja
 
-    # run joint qc and panel comparison per chromosome
+    scatter (i in range(length(vcfs))) {
+        # convert batches from vcf to bed
+        call vcf_to_bed {
+            input: vcf=vcfs[i], exclude_samples=exclude_samples[i], include_regex=include_regex, docker=docker
+        }
+        # run glm between each batch and panel
+        call glm_vs_panel {
+            input: base=sub(sub(basename(vcfs[i], ".vcf"), "_original", ""), ".calls", ""),
+            beds=[vcf_to_bed.out_bed], bims=[vcf_to_bed.out_bim], fams=[vcf_to_bed.out_fam],
```

Comment:
is glm_vs_panel used elsewhere where a list of beds is given? If not then for readability could be reduced to just ingesting single plink file.... I'm assuming there is no wdl magic here that surrounding output in a scatter from previous task with brackets does not indicate here that all of the files from previous step are going to be the input? In that case this glm_vs_panel would be joint qc across batches

Reply:
Agree but kept it more general, it works now with one dataset or many (merging is done in the beginning of the task)

Reply:
yep its fair and can be categorized as a matter of taste.  
This reminded me that we should have discussion if we'd like to have some wdl task  inline code documentation style.  Helps when either have not looked the code in a long time (like me in code review now) or when someone else then primary devs. are doing changes.